rdf4j/src/main/java/org/apache/commons/rdf/rdf4j/RDF4JQuad.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.rdf.rdf4j;
+
+import org.apache.commons.rdf.api.Quad;
+
+/**
+ * Marker interface for RDF4J implementations of Quad.
+ * 
+ */
+public interface RDF4JQuad extends Quad, RDF4JTripleLike {
```

Comment:
Both [Quad](http://stain.github.io/incubator-commonsrdf/integration/org/apache/commons/rdf/api/Quad.html) and [Triple](http://stain.github.io/incubator-commonsrdf/integration/org/apache/commons/rdf/api/Triple.html) are [TripleLike](http://stain.github.io/incubator-commonsrdf/integration/org/apache/commons/rdf/api/TripleLike.html) in CommonsRDF, they both have methods `getSubject()`, `getPredicate()` and `getObject()`.  However `Quad` and `Triple` have different equality semantics (quad also considers the graph label) - which is particularly important for a `Statement` because a Triple can be in multiple contexts.

`Quad` can't be a subtype of `Triple` as you don't get bidirectional `.equals()` (or we would have to require `Triple.equals()` to recognize `Quad`) - but they can share the "duck typing" interface `TripleLike` for users who don't care about equality, but just want to access say the `getSubject()` method. See #19 for discussion.
 
To determine if a _triple like_ is `Triple` or `Quad`, use `instanceof`. (It may be a generalized triple, which is neither).